firefox-ios/Client/Frontend/Toolbar+URLBar/OverlayModeManager.swift

```diff
@@ -67,6 +67,7 @@ class DefaultOverlayModeManager: OverlayModeManager {
     }
 
     func finishEditing(shouldCancelLoading: Bool) {
+        guard inOverlayMode else { return }
```

Comment:
This triggers an update of the entire BVC and adds a lot of noise to the logs so I added a check to only do this when needed.